modules/data.atmosphere/R/ERA5_download.R

```diff
@@ -0,0 +1,142 @@
+      "Failed to load `cdsapi` Python library. ",
+      "Please make sure it is installed to a location accessible to `reticulate`.",
+      "You should be able to install it with the following command: ",
+      "`pip install --user cdsapi`.",
+      "The following error was thrown by `reticulate::import(\"cdsapi\")`: ",
+      conditionMessage(e)
+    )
+  })
+  #define function for building credential file.
+  #maybe as a helper function.
+  getnetrc <- function (dl_dir) {
+    netrc <- file.path(dl_dir, ".cdsapirc")
+    if (file.exists(netrc) == FALSE ||
+        any(grepl("https://cds.climate.copernicus.eu/api/v2",
+                  readLines(netrc))) == FALSE) {
+      netrc_conn <- file(netrc)
+      writeLines(c(
+        sprintf(
+          "url: %s",
+          getPass::getPass(msg = "Enter URL from the following link \n (https://cds.climate.copernicus.eu/api-how-to#install-the-cds-api-key):")
+        ),
+        sprintf(
+          "key: %s",
+          getPass::getPass(msg = "Enter KEY from the following link \n (https://cds.climate.copernicus.eu/api-how-to#install-the-cds-api-key):")
+        )
+      ),
+      netrc_conn)
+      close(netrc_conn)
+      message(
+        "A netrc file with your CDS Login credentials was stored in the output directory "
+      )
+    }
+    return(netrc)
+  }
+  #check if the token exists for the cdsapi.
+  if (!file.exists(file.path(Sys.getenv("HOME"), ".cdsapirc")) & auto.create.key) {
+    getnetrc(Sys.getenv("HOME"))
+  } else if (!file.exists(file.path(Sys.getenv("HOME"), ".cdsapirc")) & !auto.create.key) {
+    PEcAn.logger::logger.severe(
+      "Please create a `${HOME}/.cdsapirc` file as described here:",
+      "https://cds.climate.copernicus.eu/api-how-to#install-the-cds-api-key ."
+    )
+  }
+  #grab the client object.
+  tryCatch({
+    c <- cdsapi$Client()
+  }, error = function(e) {
+    PEcAn.logger::logger.severe(
+      "The following error was thrown by `cdsapi$Client()`: ",
+      conditionMessage(e)
+    )
+  })
+  # loop over years.
+  nc.paths <- c()
+  for (y in years) {
+    fname <- file.path(outdir, paste0("ERA5_", y, ".grib"))
+    # start retrieving data.
+    # you need to have an account for downloaing the files
+    # Read the documantion for how to setup your account and settings before trying this
+    # https://confluence.ecmwf.int/display/CKB/How+to+download+ERA5#HowtodownloadERA5-3-DownloadERA5datathroughtheCDSAPI
+    c$retrieve(
+      'reanalysis-era5-single-levels',
+      list(
+        'product_type' = 'ensemble_members',
+        'data_format' = 'grib',
+        "download_format" = "unarchived",
+        'day' = days,
+        'time' = times,
+        'month' = months,
+        'year' = as.character(y),
+        "area" = area,
+        'variable' = variables
+      ),
+      fname
+    )
+    # convert grib to nc file.
+    nc.path <- gsub(".grib", ".nc", fname, fixed = T)
+    cmd <- paste("grib_to_netcdf", fname, "-o", nc.path)
```

Comment:
what ensures that this command line utility exists in the user's system? it's not in the DESCRIPTION file, nor do you seem to provide any check that the command exists or instructions on how to install it. If you add a check, I'd do it early in the function -- would be a pain to have this function fail AFTER you've done the file download.

Reply:
Fixed.